lib/reek/smell_warning.rb

```diff
@@ -59,13 +59,21 @@ def yaml_hash
 
     protected
 
-    def contains_all?(patterns)
-      rpt = sort_key.to_s
-      patterns.all? { |pattern| pattern =~ rpt }
-    end
-
     def sort_key
       [context, message, smell_category]
     end
+
+    def common_parameters_equal?(other_parameters)
+      # Why not check for strict parameter equality instead of just the common ones?
+      #
+      # In `self`, `parameters` might look like this:  {:name=>"@other.thing", :count=>2}
+      # Coming from specs, 'other_parameters' might look like this, e.g.:
+      # {:name=>"@other.thing"}
+      # So in this spec we are just specifying the "name" parameter but not the "count".
+      # In order to allow for this kind of leniency we just test for common parameter equality,
+      # not for a strict one.
+      common_keys = parameters.keys &  other_parameters.keys
```

Comment:
Good point. They will be ignored. What do you think about raising ArgumentError?